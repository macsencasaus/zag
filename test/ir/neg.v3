// RUN: %vector -emit-ir %s

fn i32 main() {
    i32 x = 2;
    i32 y = x;
    i32 z = -y;
    z = -z;
    return z;
}
