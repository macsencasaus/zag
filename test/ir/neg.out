; V3 Readable Intermediate Representation
; Functions contain two fields:
; - The number of parameters
; - Maximum stack size (in bytes)

main(0,16):
    %s[0] = i32 2
    %s[4] = i32 %s[0]
    %s[8] = neg i32 %s[4]
    %s[8] = i32 %s[8]
    %s[12] = neg i32 %s[8]
    %s[8] = i32 %s[12]
    ret i32 %s[8]

